src/app/zeko/zkapps_rollup.ml

```diff
@@ -402,12 +402,64 @@ module Outer = struct
 
   module Path = SnarkList (Path')
 
+  (* Pickles allows to verify max 2 proofs recursively, so we wrap the extensions into separate proof *)
```

Comment:
I really dislike this design, and it's why I moved away from it, even if we sacrifice speed potentially.
The better solution is to have a generic wrapper circuit for these cases. I'll get working on it.